packages/story-editor/src/components/canvas/displayLayer.js

```diff
@@ -152,20 +142,59 @@ function DisplayLayer() {
 
   return (
     <StoryAnimation.Provider
-      animations={currentPage?.animations}
-      elements={currentPage?.elements}
+      animations={currentPageAnimations}
+      elements={currentPageElements}
       onWAAPIFinish={resetAnimationState}
-      selectedElementIds={
-        animationState === STORY_ANIMATION_STATE.PLAYING_SELECTED
-          ? animatedElements
-          : []
-      }
+      selectedElementIds={isAnimationPlaying ? animatedElements : STABLE_ARRAY}
     >
       <DisplayPageAnimationController
-        page={currentPage}
-        animationState={animationState}
         resetAnimationState={resetAnimationState}
       />
+      {children}
+    </StoryAnimation.Provider>
+  );
+}
+
+StoryAnimations.propTypes = {
+  children: PropTypes.node,
+};
+
+function DisplayLayer() {
+  const { backgroundColor, isBackgroundSelected, pageAttachment } = useStory(
+    ({ state }) => {
+      return {
+        backgroundColor: state.currentPage?.backgroundColor,
+        isBackgroundSelected:
+          state.selectedElements?.[0]?.id ===
+          state.currentPage?.elements?.[0]?.id,
```

Comment:
💭  could check to see if `isBackground === true` is on the selected element instead of checking the currentPage, no?

Reply:
yup. good call. was just pulling this check from somewhere else. will update